banyan/src/tree.rs

```diff
@@ -223,6 +240,35 @@ impl<
         }
     }
 
+    /// Returns an iterator yielding all indexes that have values matching the
+    /// provided query.
+    pub fn iter_index(
+        &self,
+        tree: &Tree<T>,
+        query: impl Query<T> + Clone + 'static,
+    ) -> impl Iterator<Item = Result<Arc<Index<T>>>> + 'static {
+        match &tree.root {
+            Some(index) => self.index_iter0(query, index.clone()).boxed().left_iter(),
```

Comment:
I know that this is just the repetition of a pattern, but still: why not just `IndexIter::new(...).left_iter()`?